repository/kafka/operator/templates/bootstrap.yaml

```diff
@@ -9,35 +9,69 @@ data:
     cp /health-check-script/health-check.sh health-check.sh;
     chmod +x health-check.sh;
 
+    {{ if eq .Params.TRANSPORT_ENCRYPTION_ENABLED "true" }}
+    /etc/tls/bin/enable-tls.sh
+    {{ end }}
+
     {{ if eq .Params.KERBEROS_ENABLED "true" }}
-    
     cat /kafka-keytab/kafka.keytab | base64 --decode > kafka.keytab;
     cp /jass-config/kafka_server_jaas.conf $KAFKA_HOME/config/kafka_server_jaas.conf;
     cp /krb5-config/krb5.conf $KAFKA_HOME/config/krb5.conf;
     sed -i "s/<HOSTNAME>/$(hostname -f)/g" $KAFKA_HOME/config/kafka_server_jaas.conf;
     export KAFKA_OPTS="-Djava.security.auth.login.config=${KAFKA_HOME}/config/kafka_server_jaas.conf -Djava.security.krb5.conf=${KAFKA_HOME}/config/krb5.conf $KAFKA_OPTS"
-    
     {{ if eq .Params.KERBEROS_DEBUG "true" }}
     export KAFKA_OPTS="-Dsun.security.krb5.debug=true $KAFKA_OPTS"
     {{ end }}
-    
     {{ end }}
     
     KAFKA_BROKER_ID=${HOSTNAME##*-}
     
-    # LISTENERS CONFIGURATION
-    LISTENERS="INTERNAL://0.0.0.0:${KAFKA_BROKER_PORT}"
-    # ADVERTISED LISTENERS
-    ADVERTISED_LISTENERS="INTERNAL://$(hostname -f):${KAFKA_BROKER_PORT}"
+    LISTENERS=()
+    ADVERTISED_LISTENERS=()
+
     {{ if eq .Params.KERBEROS_ENABLED "true" }}
+    # Kerberos enabled
+    {{ if eq .Params.TRANSPORT_ENCRYPTION_ENABLED "true" }}
+    # Transport encryption on
+    LISTENERS=("${LISTENERS[@]}" "INTERNAL://0.0.0.0:${KAFKA_BROKER_PORT_TLS}")
+    ADVERTISED_LISTENERS=("${ADVERTISED_LISTENERS[@]}" "INTERNAL://$(hostname -f):${KAFKA_BROKER_PORT_TLS}")
+    {{ if eq .Params.TRANSPORT_ENCRYPTION_ALLOW_PLAINTEXT "true" }}
```

Comment:
I wonder if we should consider indenting these `if`s to visually be able to separate nested if blocks?

Reply:
It will cause the commands in the rendered `bootstrap.sh` script to be oddly indented. I had a discussion with @zmalik and opted for better user experience vs better developer experience (making debugging in the launched containers easier).